src/utils/netlify-utils.ts

```diff
@@ -0,0 +1,98 @@
+import axios, { AxiosError } from "axios"
+
+import { config } from "@config/config"
+
+import logger from "@root/logger/logger"
+
+const GITHUB_ORG_NAME = config.get("github.orgName")
+const NETLIFY_ACCESS_TOKEN = config.get("netlify.accessToken")
+
+type NetlifySiteDetails = {
+  id: string
+  password: string
+  build_settings: {
+    repo_path: string
+    repo_branch: string
+  }
+}
+
+type UpdateNetlifySiteDetails = {
+  password: string
+  build_settings: {
+    stop_builds: true
+  }
+}
+
+// Retrieves all netlify sites which build from the given repo's staging branch
+const getNetlifySiteDetails = async (
+  repoName: string
+): Promise<NetlifySiteDetails[]> => {
+  const endpoint = "https://api.netlify.com/api/v1/sites"
+  const headers = {
+    Authorization: `Bearer ${NETLIFY_ACCESS_TOKEN}`,
+    "Content-Type": "application/json",
+  }
+  const repoPath = `${GITHUB_ORG_NAME}/${repoName}`
+  try {
+    const allSites = await axios.get<NetlifySiteDetails[]>(endpoint, {
+      headers,
+    })
+    const relatedSites = allSites.data.filter(
+      (site) =>
+        site.build_settings.repo_path === repoPath &&
+        site.build_settings.repo_branch === "staging"
```

Comment:
what about for prod builds?

Reply:
Prod is okay - we're fine with keeping prod up to date, since the issue is with embargoed information on staging

Reply:
oh true good point